dashboard/src/components/AppView/AppView.tsx

```diff
@@ -191,6 +193,16 @@ export default function AppView() {
     resourceRefs;
   const revision = app?.revision ?? 0;
   const icon = appDetails?.iconUrl ?? placeholder;
+
+  // If the package identifier matches the current list of loaded customAppViews,
+  // then load the custom view from external bundle instead of the default one.
+  if (
+    app?.availablePackageRef?.identifier &&
+    customAppViews.includes(app.availablePackageRef.identifier)
```

Comment:
I'm doubtful about the proper way to match a given package to its associated view. Not sure if the identifier is the best way.

Reply:
+1. Yes, sorry for not providing input earlier (we had a crazy couple of weeks), but I'm a little concerned about adding this in a way that won't be compatible when we're enabling support for pluggable packaging formats. At the very least, we'd need the plugin being used (`helm`) as well as the (plugin-specific) identifier proposed above.

That is, to future-proof this, we need to ensure the `customAppViews` data isn't just string, but structured data with a couple of fields: `{ "plugin": "helm", "name": "apache" }`. Note that this would be the `apache` chart from any repository. That would also enable later adding extra fields for more granular support (such as `{ "plugin": "helm", "name": "apache", "repository": "bitnami", "namespace": "kubeapps"}` etc.)

Is that an option? (for now, each custom app view defining both the `plugin` (just `helm`) and the name?

Thanks for the updates based on the recent changes Antonio, and thanks Anthony for the contributions.

Reply:
With the new changes and functionality that you guys want to add this is a good point. At the moment I'm getting antiquated w/ some of the interface changes and I'll figure out a way to target more specific deployments